pkg/util/encoding_test.go

```diff
@@ -130,3 +131,89 @@ func TestDecodeQuotedPrintable(t *testing.T) {
 		assert.Equal(t, str_out, val)
 	})
 }
+
+func TestGetRandomString(t *testing.T) {
+	charset := "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"
+	chars := len(charset)
+	length := 20
+	rounds := 50_000
+
+	// Generate random strings and count the frequency of each character
+	m := make(map[string]int)
+	for i := 0; i < rounds; i++ {
+		r, err := GetRandomString(length)
+		require.NoError(t, err)
+
+		for _, c := range r {
+			m[string(c)]++
+		}
+	}
+
+	// Find lowest and highest frequencies
+	min := rounds * length
+	max := 0
+
+	// Calculate chi-squared statistic
+	expected := float64(rounds) * float64(length) / float64(chars)
+	chiSquared := 0.0
+
+	for _, char := range charset {
+		if m[string(char)] < min {
+			min = m[string(char)]
+		}
+		if m[string(char)] > max {
+			max = m[string(char)]
+		}
+		chiSquared += math.Pow(float64(m[string(char)])-expected, 2) / expected
+	}
+
+	// Ensure there is no more than 5% variance between lowest and highest frequency characters
+	assert.LessOrEqual(t, float64(max-min)/float64(min), 0.05, "Variance between lowest and highest frequency characters must be no more than 5%")
+
+	// Ensure chi-squared value is lower than the critical bound
+	// 99.9% probability for 61 degrees of freedom
+	assert.Less(t, chiSquared, 100.888, "Chi squared value must be less than the 99.9% critical bound")
```

Comment:
I tried to run this 10 000 times (it timed out after 10 minutes, I'm a little unsure how many rounds it got through, but since 1000 runs took a little over a minute, I don't think the lost runs on the end are too numerous) and it failed 165 times for the variance check and 3 times for the chi check, for a little over 1.5% failure rate.

I think that's a bit too frequent.

Reply:
I'll tweak the tests to make them a bit less sensitive, they're probably tighter than they need to be right now.

Reply:
I relaxed the test criteria to 10% variance and 99.99% chi squared, 5k tests without any failures.  The old code fails both chi squared and the string variance test.  It passes the digits variance test which is expected because the skew was less than 10% when generating digits.